File: src/Services/WareHouse/WareHouse.API/Application/Model/WareHouseDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WareHouse.API.Application.Model
{
    public class WareHouseDTO: BaseModel
    {
        public string Code { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Description { get; set; }
        public string ParentId { get; set; }
        public string Path { get; set; }
        public bool Inactive { get; set; }
    }
}

File: src/WebClient/WareHouse/src/app/model/WareHouseDTO.ts
import { BaseModel } from "./BaseModel";

export interface WareHouseDTO extends BaseModel {
    code: string;
    name: string;
    address: string;
    description?: any;
    parentId: string;
    path?: any;
    inactive: boolean;
    id: string;
    wareHouseDTOs:WareHouseDTO[];
}